dll/shellext/cabview/util.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * PROJECT:     ReactOS CabView Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Utility header file
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#pragma once
+
+template<class H> static int ErrorBox(H hWnd, int Error)
+{
+    SHELL_ErrorBox(hWnd, Error);
+    return Error;
+}
+
+template<class T> static inline bool IsPathSep(T c)
+{
+    return c == '\\' || c == '/';
+}
+
+inline bool IsEqual(const SHCOLUMNID &scid, REFGUID guid, UINT pid)
+{
+    return scid.pid == pid && IsEqualGUID(scid.fmtid, guid);
+}
+
+inline HRESULT InitVariantFromBuffer(const void *buffer, UINT cb, VARIANT *pv)
+{
+    SAFEARRAY *pa = SafeArrayCreateVector(VT_UI1, 0, cb);
+    if (pa)
+    {
+        CopyMemory(pa->pvData, buffer, cb);
+        V_VT(pv) = VT_UI1 | VT_ARRAY;
+        V_UNION(pv, parray) = pa;
+        return S_OK;
+    }
+    V_VT(pv) = VT_EMPTY;
+    return E_OUTOFMEMORY;
+}
+
+inline void FreeStrRet(STRRET &str)
+{
+    if (str.uType == STRRET_WSTR)
+    {
+        SHFree(str.pOleStr);
+        str.uType = STRRET_CSTR;
+    }
+}
+
+inline HRESULT StrTo(LPCWSTR str, UINT len, STRRET &sr)
```

Comment:
We have `SHSetStrRet` in `reactos/shellutils.h`

Reply:
But not one that takes a length as input (I'm chopping off the filename from the path in folder.cpp).

Reply:
`StrTo` is overloaded on the output parameter, it will put a string in a `STRRET` or a `VARIANT`. This folder implementation fully supports both `GetDetailsOf` and `GetDetailsEx` unlike the other folders in ROS.